File: app/src/main/java/com/elfefe/wiid/controllers/fragments/MapsFragment.java
package com.elfefe.wiid.controllers.fragments;

import android.content.Context;
import android.location.Address;
import android.location.Geocoder;
import android.os.Bundle;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import com.elfefe.wiid.R;
import com.google.android.gms.maps.*;
import com.google.android.gms.maps.model.LatLng;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class MapsFragment extends Fragment implements OnMapReadyCallback {

        private GoogleMap map;
        private MapView mapView;
        private EditText search;
        private Context context;

    public static Fragment newInstance(){
        return new MapsFragment();
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState){
        View view = inflater.inflate(R.layout.fragment_maps, container, false);
        mapView = view.findViewById(R.id.fragment_maps_mapView);
        search = view.findViewById(R.id.fragment_maps_search);

        context = getActivity();

        mapView.onCreate(savedInstanceState);
        mapView.getMapAsync(this);

        return view;
    }

    @Override
    public void onMapReady(GoogleMap googleMap) {
        map = googleMap;
    }

    @Override
    public void onResume() {
        mapView.onResume();
        super.onResume();
        search.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {
                List<Address> locations = new ArrayList<>();
                if (!search.getText().toString().equals("")) {
                    Geocoder geocoder = new Geocoder(context);
                    try {
                        if(geocoder.getFromLocationName(search.getText().toString().toLowerCase(), 1) != null) {
                            locations = geocoder.getFromLocationName(search.getText().toString().toLowerCase(), 1);
                        }
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                }
                if (locations != null) {
                    if (locations.size() > 0) {
                        Address location = locations.get(0);
                        LatLng localisation = new LatLng(location.getLatitude(), location.getLongitude());
                        map.animateCamera(CameraUpdateFactory.newLatLngZoom(localisation, 10f));
                    }
                }
            }
        });
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mapView.onDestroy();
    }

    @Override
    public void onLowMemory() {
        super.onLowMemory();
        mapView.onLowMemory();
    }
}

File: app/src/main/java/com/elfefe/wiid/controllers/fragments/MapsFragment.kt
package com.elfefe.wiid.controllers.fragments

import android.location.Address
import android.location.Geocoder
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import com.elfefe.wiid.R
import com.google.android.gms.maps.*
import com.google.android.gms.maps.model.LatLng

class MapsFragment : Fragment(), OnMapReadyCallback {

    lateinit var map: GoogleMap
    lateinit var mapView: MapView
    lateinit var search: EditText

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_maps, container, false)

        mapView = view.findViewById(R.id.fragment_maps_mapView)
        search = view.findViewById(R.id.fragment_maps_search)
        mapView.onCreate(savedInstanceState)

        mapView.getMapAsync(this)

        return view
    }

    override fun onMapReady(googleMap: GoogleMap) {
        map = googleMap
    }

    override fun onResume() {
        mapView.onResume()
        super.onResume()
        search.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable) {
                    if (search.text.toString().toLowerCase() == "amsterdam") {

                        val location: Address = Geocoder(context).getFromLocationName("Amsterdam", 1).get(0)
                        val localisation = LatLng(location.latitude, location.longitude)
                        map.animateCamera(CameraUpdateFactory.newLatLngZoom(localisation, 10f))
                    }
            }
        })

    }

    override fun onDestroy() {
        super.onDestroy()
        mapView.onDestroy()
    }

    override fun onLowMemory() {
        super.onLowMemory()
        mapView.onLowMemory()
    }

    companion object {

        fun newInstance(): MapsFragment {
            return MapsFragment()
        }
    }
}